homeassistant/components/modbus/__init__.py

```diff
@@ -136,7 +137,11 @@ def start_modbus():
         await hub_collect[client_name].write_coil(unit, address, state)
 
     # do not wait for EVENT_HOMEASSISTANT_START, activate pymodbus now
-    await hass.async_add_executor_job(start_modbus)
+    for client in hub_collect.values():
+        await client.setup(hass)
```

Comment:
Does each client setup need to run sequentially or can we use asyncio to make them run concurrent? Eg with `asyncio.gather`.

Reply:
We can use asyncio.gather, however I would really prefer to do that in a later PR, in order only to fix the problem at hand here.

If OK, I promise to make a new PR (after I have solved the outstanding problems).

Reply:
Sure! :+1: